sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -190,6 +192,28 @@ def predict(self, X):
         return np.ones(X.shape[0])
 
 
+class BadBalancedWeightsClassifier(BaseBadClassifier):
+    def __init__(self, class_weight=None):
+        self.class_weight = class_weight
+
+    def fit(self, X, y):
+        from sklearn.preprocessing import LabelEncoder
+        from sklearn.utils import compute_class_weight
+
+        label_encoder = LabelEncoder().fit(y)
+        classes = label_encoder.classes_
+        class_weight = compute_class_weight(self.class_weight, classes, y)
+
+        # Intentionally modify the balanced class_weight
+        # to simulate a bug and raise an exception
+        if self.class_weight == "balanced":
+            class_weight += 1.
```

Comment:
I don't get what this is doing. Surely this doesn't result in affecting the output of `predict`. In fact the reason why the test is passing may be because predict isn't affected by changing class weight at all...

Reply:
No, you are right, this dummy class does not affect the output of `predict`. Instead, it affects `class_weight`. 

I did it that way because, as I understand, the check `check_class_weight_balanced_linear_classifier` this test is testing does not verify the output of `predict`. It simply verifies `coef_` after the estimator has been fitted. 
https://github.com/scikit-learn/scikit-learn/blob/bf6949ba9d5fc019fd48eda247249d8fd8c420fa/sklearn/utils/estimator_checks.py#L1994